packages/calypso-e2e/src/lib/components/full-side-editor-data-views-component.ts

```diff
@@ -30,6 +30,13 @@ export class FullSiteEditorDataViewsComponent {
 	 */
 	async clickPrimaryFieldByExactText( primaryFieldText: string ): Promise< void > {
 		const editorParent = await this.editor.parent();
-		await editorParent.locator( selectors.primaryFieldByText( primaryFieldText ) ).click();
+		const primaryFieldInTable = editorParent.locator(
+			selectors.primaryFieldByText( 'table', primaryFieldText )
+		);
+		const primaryFieldInGrid = editorParent.locator(
+			selectors.primaryFieldByText( 'grid', primaryFieldText )
+		);
+
+		await Promise.race( [ primaryFieldInTable.click(), primaryFieldInGrid.click() ] );
```

Comment:
Looks a bit weird, could you please elaborate why we have two locators and finishing only one one of them is resolved?

Reply:
That's because the selectors vary between versions of GB, and platforms as well. In some instances a grid is expected. Other versions expect a table etc.